web/packages/teleterm/src/ui/DocumentCluster/UnifiedResources.tsx

```diff
@@ -257,8 +315,20 @@ const Resources = memo(
           props.updateUserPreferences({ unifiedResourcePreferences })
         }
         pinning={pinning}
-        resources={resources.map(mapToSharedResource)}
-        resourcesFetchAttempt={attempt}
+        resources={resources.map(r => {
+          const { resource, ui } = mapToSharedResource(r);
+          return {
+            resource,
+            ui: {
+              ActionButton: props.getAccessRequestButton(r) || ui.ActionButton,
+            },
+          };
+        })}
+        resourcesFetchAttempt={
+          props.showResources === ShowResources.UNSPECIFIED
+            ? { status: 'processing' }
+            : attempt
+        }
```

Comment:
So now if Connect fails to fetch cluster details, it is going to show a permanent spinner, isn't it? Is there a better way to address this error case?

Reply:
Good catch. As discussed in DMs, adding a proper fetch status to the clusters service requires significant refactor. 
Instead, I decided to reset the `showResources` field to `ShowResources.ACCESSIBLE_ONLY` if we fail to fetch details.